Проектная работа №1 «Калькулятора счёта».

AnShok opened the pull request:
Сдача на ревью кода java «Калькулятора счёта».
Добавлен класс "Calculator", содержащий логику добавления и подсчёта товаров. 
Добавлен класс "Product" с полями "имя" — строка и "цена" — дробное значение.
Добавлен класс "Formatter", содержащий метод получения слова рубль в правильном падеже.

Commit:
Сдача на ревью кода java «Калькулятора счёта».

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,32 @@
+        int numberOfGuests;
+
+        while (true) {
+            System.out.println("На скольких человек необходимо разделить счёт.: ");
```
Стилистически тут лучше поставить вопросительный знак в конце

AleksandrIlinskii commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,76 @@
+
+
+    public Calculator() {
+        products = new Product[100];
```
Если ввести 101 значение - мы получим 
<img width="727" alt="image" src="https://user-images.githubusercontent.com/129396632/233332271-d1b00ae6-8eb0-4f95-862f-2621c84a704f.png">
И краш программы

AleksandrIlinskii replied:
Тут цифра 3, потому что я поменял с 100 на 3, так как вводить долго. Но такое теоретически возможно

AleksandrIlinskii commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,76 @@
+        String formattedTotalCost = Formatter.formatTotalCost(totalCost);
+        System.out.println("Общая стоимость товаров: " + formattedTotalCost);
+    }
+    public void run() {
```
Судя по названию метода - надо бежать. Лучше называть по смыслу того, что происходит внутри или возвращает

Commit:
Исправлены недочеты:
1. Фиксированный массив заменен на ArrayList, что не ограничивает в колличестве товаров.
2. Исправлена стилистика 1-ого вопроса.
3. Переименован метод "run()" - makeBasketProducts()

AnShok commented:
Исправлены недочеты и краш программы при неизвестном количестве товаров.
